server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -1251,6 +1256,12 @@ public SnapshotInfo takeSnapshot(VolumeInfo volume) throws ResourceAllocationExc
 
             SnapshotInfo snapshotOnPrimary = snapshotStrategy.takeSnapshot(snapshot);
             boolean backupSnapToSecondary = SnapshotInfo.BackupSnapshotAfterTakingSnapshot.value() == null || SnapshotInfo.BackupSnapshotAfterTakingSnapshot.value();
+            if (backupSnapToSecondary) {
+                DataCenterVO zone = dataCenterDao.findById(snapshot.getDataCenterId());
+                if (DataCenter.Type.Edge.equals(zone.getType())) {
+                    backupSnapToSecondary = false;
+                }
+            }
```

Comment:
@shwstppr can you put this in a separate method?

Reply:
done